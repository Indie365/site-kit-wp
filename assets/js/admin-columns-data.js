/**
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// This file should not use any dependencies because it is used in the frontend.

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';

/**
 * Fetches the current post data for custom columns.
 *
 * @since n.e.x.t
 */
async function fetchColumnsData() {
	const columns = document.querySelectorAll( 'td.column-label' );
	const filteredColumns = {};

	for ( const column of columns ) {
		if ( ! filteredColumns[ column.getAttribute( 'data-colname' ) ] ) {
			filteredColumns[ column.getAttribute( 'data-colname' ) ] = column;
		}
	}

	for ( const column in filteredColumns ) {
		const columnName =
			filteredColumns[ column ].getAttribute( 'data-colname' );
		const postIDs = [];

		const posts = document.querySelectorAll(
			`[data-colname=${ columnName }] .googlesitekit-views-column`
		);

		for ( const post of posts ) {
			postIDs.push( post.getAttribute( 'data-id' ) );
		}

		const data = await API.get(
			'core',
			'admin',
			'columns-data',
			{
				data: {
					post_type: 'post',
					posts: postIDs,
				},
			},
			{ useCache: false }
		);

		if ( data ) {
			for ( const postID in data ) {
				const viewSpan = document.querySelector(
					`.googlesitekit-views-column[data-id="${ postID }"]`
				);
				if ( viewSpan ) {
					viewSpan.textContent = data[ postID ].views;
				}
			}
		}
	}
}

fetchColumnsData();
