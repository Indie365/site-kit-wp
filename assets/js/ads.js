/**
 * This file is intended to detect active ad blocker.
 *
 * Ad blockers block URLs containing the word "ads" including this file.
 * If the file does load, `googlesitekit.canAdsRun` is set to true.
 */

if ( global.googlesitekit === undefined ) {
	global.googlesitekit = {};
}

global.googlesitekit.canAdsRun = true;

// Ensure that this flag does not get wiped at a later stage during pageload.
document.addEventListener( 'DOMContentLoaded', function() {
	global.googlesitekit.canAdsRun = true;
} );
