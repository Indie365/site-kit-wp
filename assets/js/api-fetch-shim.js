/**
 * WordPress dependencies
 */
import apiFetch from '@wordpress/api-fetch__non-shim';

if ( global._googlesitekitAPIFetchData ) {
	if ( global._googlesitekitAPIFetchData.rootURL ) {
		apiFetch.use( apiFetch.createRootURLMiddleware( global._googlesitekitAPIFetchData.rootURL ) );
	}

	if ( global._googlesitekitAPIFetchData.nonceMiddleware ) {
		apiFetch.nonceMiddleware = apiFetch.createNonceMiddleware();
		apiFetch.use( apiFetch.createNonceMiddleware( global._googlesitekitAPIFetchData.nonceMiddleware ) );
		apiFetch.use( apiFetch.mediaUploadMiddleware );
		apiFetch.nonceEndpoint = global._googlesitekitAPIFetchData.nonceEndpoint;
	}
}

export * from '@wordpress/api-fetch__non-shim';

export default apiFetch;
