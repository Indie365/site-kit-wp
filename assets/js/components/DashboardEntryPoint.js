/**
 * DashboardEntryPoint component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { useFeature } from '../hooks/useFeature';
import ModuleSetup from './setup/ModuleSetup';
import DashboardApp from './dashboard/DashboardApp';
import DashboardMainApp from './DashboardMainApp';
import NotificationCounter from './legacy-notifications/notification-counter';

export default function DashboardEntryPoint( { setupModuleSlug } ) {
	const unifiedDashboardEnabled = useFeature( 'unifiedDashboard' );

	if ( !! setupModuleSlug ) {
		return <ModuleSetup moduleSlug={ setupModuleSlug } />;
	}

	if ( unifiedDashboardEnabled ) {
		return <DashboardMainApp />;
	}

	return (
		<Fragment>
			<NotificationCounter />
			<DashboardApp />
		</Fragment>
	);
}

DashboardEntryPoint.propTypes = {
	setupModuleSlug: PropTypes.string,
};
