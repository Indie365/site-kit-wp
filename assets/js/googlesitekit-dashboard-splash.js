/**
 * DashboardSplash component.
 *
 * Site Kit by Google, Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { clearWebStorage, loadTranslations } from 'GoogleUtil';
import Notification from 'GoogleComponents/notifications/notification';
import 'GoogleComponents/notifications';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { Component, render, Fragment } from '@wordpress/element';
import { doAction } from '@wordpress/hooks';

/**
 * Internal dependencies
 */
import DashboardSplashApp from './components/dashboard-splash/dashboard-splash-app';
import NotificationCounter from './components/notifications/notification-counter';

class GoogleSitekitDashboardSplash extends Component {
	constructor( props ) {
		super( props );

		this.state = {
			hasError: false,
		};
	}

	componentDidCatch( error, info ) {
		this.setState( {
			hasError: true,
			error,
			info,
		} );
	}

	render() {
		const {
			hasError,
			error,
			info,
		} = this.state;

		if ( hasError ) {
			return <Notification
				id={ 'googlesitekit-error' }
				key={ 'googlesitekit-error' }
				title={ error.message }
				description={ info.componentStack }
				dismiss={ '' }
				isDismissable={ false }
				format="small"
				type="win-error"
			/>;
		}

		return (
			<Fragment>
				<NotificationCounter />
				<DashboardSplashApp />
			</Fragment>
		);
	}
}

// Initialize the app once the DOM is ready.
domReady( () => {
	if ( global.googlesitekit.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById( 'js-googlesitekit-dashboard-splash' );

	if ( renderTarget ) {
		loadTranslations();

		render( <GoogleSitekitDashboardSplash />, renderTarget );

		/**
		 * Action triggered when the Dashboard Splash App is loaded.
		 */
		doAction( 'googlesitekit.moduleLoaded', 'Splash' );
	}
} );
