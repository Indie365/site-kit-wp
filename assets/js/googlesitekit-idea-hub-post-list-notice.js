/**
 * Idea Hub Post List notice.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import { CORE_USER } from './googlesitekit/datastore/user/constants';
const { dispatch } = Data;

const WEEK_IN_SECONDS = 3600 * 24 * 7;

domReady( () => {
	const notice = document.querySelector(
		'[id="googlesitekit-notice-idea-hub_new-ideas"],[id="googlesitekit-notice-idea-hub_saved-ideas"]'
	);
	if ( ! notice ) {
		return;
	}
	const type = notice.id.replace( 'googlesitekit-notice-', '' );
	const expiresInSeconds =
		type === 'idea-hub_new-ideas' ? WEEK_IN_SECONDS : 0;

	notice.addEventListener( 'click', ( event ) => {
		if ( event.target.classList.contains( 'notice-dismiss' ) ) {
			dispatch( CORE_USER ).dismissItem( type, { expiresInSeconds } );
		}
	} );
} );
