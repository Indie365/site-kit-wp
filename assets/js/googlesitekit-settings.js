/**
 * Settings component.
 *
 * Site Kit by Google, Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* eslint camelcase:[0] */

/**
 * External dependencies
 */
import SettingsApp from 'GoogleComponents/settings/settings-app';
import Notification from 'GoogleComponents/notifications/notification';
import 'GoogleComponents/notifications';
import { loadTranslations } from 'GoogleUtil';
import 'GoogleModules';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { doAction } from '@wordpress/hooks';
import { Component, render } from '@wordpress/element';

class GoogleSitekitSettings extends Component {
	constructor( props ) {
		super( props );

		this.state = {
			hasError: false,
		};
	}

	componentDidCatch( error, info ) {
		this.setState( {
			hasError: true,
			error,
			info,
		} );
	}

	render() {
		const {
			hasError,
			error,
			info,
		} = this.state;

		if ( hasError ) {
			return <Notification
				id={ 'googlesitekit-error' }
				key={ 'googlesitekit-error' }
				title={ error.message }
				description={ info.componentStack }
				dismiss={ '' }
				isDismissable={ false }
				format="small"
				type="win-error"
			/>;
		}

		return <SettingsApp />;
	}
}

// Initialize the app once the DOM is ready.
domReady( () => {
	const renderTarget = document.getElementById( 'googlesitekit-settings-wrapper' );

	if ( renderTarget ) {
		loadTranslations();

		render( <GoogleSitekitSettings />, renderTarget );

		/**
		 * Action triggered when the settings App is loaded.
		 */
		doAction( 'googlesitekit.moduleLoaded', 'Settings' );
	}
} );
