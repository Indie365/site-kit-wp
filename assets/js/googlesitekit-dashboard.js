/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render, Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { clearWebStorage } from './util';
import { useFeature } from './hooks/useFeature';
import Root from './components/Root';
import ModuleSetup from './components/setup/ModuleSetup';
import DashboardApp from './components/dashboard/DashboardApp';
import DashboardMainApp from './components/DashboardMainApp';
import NotificationCounter from './components/legacy-notifications/notification-counter';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';

const GoogleSitekitDashboard = ( { setupModuleSlug } ) => {
	const unifiedDashboardEnabled = useFeature( 'unifiedDashboard' );

	if ( unifiedDashboardEnabled ) {
		return <DashboardMainApp />;
	}

	if ( !! setupModuleSlug ) {
		return <ModuleSetup moduleSlug={ setupModuleSlug } />;
	}

	return (
		<Fragment>
			<NotificationCounter />
			<DashboardApp />
		</Fragment>
	);
};

// Initialize the app once the DOM is ready.
domReady( () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		const { setupModuleSlug } = renderTarget.dataset;

		render(
			<Root
				viewContext={
					setupModuleSlug
						? VIEW_CONTEXT_MODULE_SETUP
						: VIEW_CONTEXT_DASHBOARD
				}
			>
				<GoogleSitekitDashboard setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
