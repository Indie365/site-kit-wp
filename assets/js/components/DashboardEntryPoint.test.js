/**
 * DashboardEntryPoint tests.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { render } from '../../../tests/js/test-utils';
import { mockCreateComponent } from '../../../tests/js/mock-component-utils';
import DashboardEntryPoint from './DashboardEntryPoint';

jest.mock( './setup/ModuleSetup', () => mockCreateComponent( 'ModuleSetup' ) );
jest.mock( './dashboard/DashboardApp', () =>
	mockCreateComponent( 'DashboardApp' )
);
jest.mock( './DashboardMainApp', () =>
	mockCreateComponent( 'DashboardMainApp' )
);
jest.mock( './legacy-notifications/notification-counter', () =>
	mockCreateComponent( 'NotificationCounter' )
);

describe( 'DashboardEntryPoint', () => {
	const unifiedDashboardRenderOptions = { features: [ 'unifiedDashboard' ] };

	it( 'should render the unified dashboard when unified dashboard is enabled', () => {
		const { container } = render(
			<DashboardEntryPoint />,
			unifiedDashboardRenderOptions
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the module setup component when unified dashboard is enabled and passed the setupModuleSlug prop', () => {
		const { container } = render(
			<DashboardEntryPoint setupModuleSlug="analytics" />,
			unifiedDashboardRenderOptions
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the non-unified dashboard when unified dashboard is not enabled', () => {
		const { container } = render( <DashboardEntryPoint /> );
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the module setup component when unified dashboard is not enabled and passed the setupModuleSlug prop', () => {
		const { container } = render(
			<DashboardEntryPoint setupModuleSlug="analytics" />
		);
		expect( container ).toMatchSnapshot();
	} );
} );
