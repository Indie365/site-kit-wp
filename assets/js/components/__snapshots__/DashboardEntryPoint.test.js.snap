// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardEntryPoint should render the module setup component when unified dashboard is enabled and passed the setupModuleSlug prop 1`] = `
<div>
  <div>
    ModuleSetup
    {"moduleSlug":"analytics"}
  </div>
</div>
`;

exports[`DashboardEntryPoint should render the module setup component when unified dashboard is not enabled and passed the setupModuleSlug prop 1`] = `
<div>
  <div>
    ModuleSetup
    {"moduleSlug":"analytics"}
  </div>
</div>
`;

exports[`DashboardEntryPoint should render the non-unified dashboard when unified dashboard is not enabled 1`] = `
<div>
  <div>
    NotificationCounter
    {}
  </div>
  <div>
    DashboardApp
    {}
  </div>
</div>
`;

exports[`DashboardEntryPoint should render the unified dashboard when unified dashboard is enabled 1`] = `
<div>
  <div>
    DashboardMainApp
    {}
  </div>
</div>
`;
